cli/src/pcluster/models/cluster_config.py

```diff
@@ -0,0 +1,1188 @@
+
+    def _validate(self):
+        self._execute_validator(
+            DuplicateInstanceTypeValidator,
+            instance_type_list=self.instance_type_list,
+        )
+        for compute_resource in self.compute_resources:
+            if compute_resource.efa:
+                self._execute_validator(
+                    EfaSecurityGroupValidator,
+                    efa_enabled=compute_resource.efa,
+                    security_groups=self.networking.security_groups,
+                    additional_security_groups=self.networking.additional_security_groups,
+                )
+                if self.networking.placement_group:
+                    self._execute_validator(
+                        EfaPlacementGroupValidator,
+                        efa_enabled=compute_resource.efa,
+                        placement_group_id=self.networking.placement_group.id,
+                        placement_group_enabled=self.networking.placement_group.enabled,
+                    )
+
+    @property
+    def instance_type_list(self):
+        """Return the list of instance types associated to the Queue."""
+        return [compute_resource.instance_type for compute_resource in self.compute_resources]
+
+
+class Dns(Resource):
+    """Represent the DNS settings."""
+
+    def __init__(self, disable_managed_dns: bool = None, domain: str = None, hosted_zone_id: str = None):
+        super().__init__()
+        self.disable_managed_dns = Resource.init_param(disable_managed_dns, default=False)
+        self.domain = Resource.init_param(domain)
+        self.hosted_zone_id = Resource.init_param(hosted_zone_id)
+
+
+class SlurmSettings(CommonSchedulingSettings):
+    """Represent the Slurm settings."""
+
+    def __init__(self, scaledown_idletime: int, dns: Dns = None):
+        super().__init__(scaledown_idletime)
+        self.dns = dns
+
+
+class SlurmScheduling(Resource):
+    """Represent a slurm Scheduling resource."""
+
+    def __init__(self, queues: List[SlurmQueue], settings: SlurmSettings = None):
+        super().__init__()
+        self.scheduler = "slurm"
+        self.queues = queues
+        self.settings = settings
+
+
+class SlurmClusterConfig(BaseClusterConfig):
+    """Represent the full Slurm Cluster configuration."""
+
+    def __init__(self, scheduling: SlurmScheduling, **kwargs):
+        super().__init__(**kwargs)
+        self.scheduling = scheduling
+
+    def _validate(self):
+        super()._validate()
+        self._execute_validator(SchedulerOsValidator, scheduler=self.scheduling.scheduler, os=self.image.os)
+
+        for queue in self.scheduling.queues:
+            for compute_resource in queue.compute_resources:
+                self._execute_validator(
+                    InstanceArchitectureCompatibilityValidator,
+                    instance_type=compute_resource.instance_type,
+                    architecture=self.head_node.architecture,
+                )
+
+
+class ClusterBucket:
```

Comment:
Why put the class here instead of cluster.py? Or should it inheritate from Resource?